module_utils/network_lsr/argument_validator.py

```diff
@@ -1152,21 +1152,263 @@ class ArgValidator_DictBond(ArgValidatorDict):
+                ArgValidatorNum(
+                    "updelay", val_min=0, val_max=1000000, default_value=None
+                ),
+                ArgValidatorNum(
+                    "use_carrier", val_min=0, val_max=1, default_value=None
+                ),
+                ArgValidatorStr(
+                    "xmit_hash_policy",
+                    enum_values=ArgValidator_DictBond.VALID_XMIT_HASH_POLICIES,
+                ),
             ],
             default_value=ArgValidator.MISSING,
         )
 
+    def _validate_post(self, value, name, result):
+        AD_OPTIONS = [
+            "ad_actor_sys_prio",
+            "ad_actor_system",
+            "ad_user_port_key",
+            "lacp_rate",
+        ]
+        ARP_OPTIONS = ["arp_interval", "arp_ip_target", "arp_validate"]
+        ARP_ONLY_MODE = ["balance-rr", "active-backup", "balance-xor", "broadcast"]
+
+        if result["mode"] != "802.3ad":
+            for option in AD_OPTIONS:
+                if result[option] is not None:
+                    raise ValidationError(
+                        name,
+                        "the bond option {0} is only valid with mode 802.3ad".format(
+                            option
+                        ),
+                    )
+
+        if result["packets_per_port"] is not None and result["mode"] != "balance-rr":
+            raise ValidationError(
+                name,
+                "the bond option packets_per_port is only valid with mode balance-rr",
+            )
+
+        if result["mode"] not in ARP_ONLY_MODE:
+            for option in ARP_OPTIONS:
+                if result[option] is not None:
+                    raise ValidationError(
+                        name,
+                        "the bond option {0} is only valid with mode balance-rr, active-backup, balance-xor or broadcast".format(
+                            option
+                        ),
+                    )
+
+        if result["tlb_dynamic_lb"] is not None and result["mode"] not in [
+            "balance-tlb",
+            "balance-alb",
+        ]:
+            raise ValidationError(
+                name,
+                "the bond option tlb_dynamic_lb is only valid with mode balance-tlb or balance-alb",
+            )
+
+        if result["primary"] is not None and result["mode"] not in [
+            "active-backup",
+            "balance-tlb",
+            "balance-alb",
+        ]:
+            raise ValidationError(
+                name,
+                "the bond option primary is only valid with mode active-backup, balance-tlb, balance-alb",
+            )
+
+        if (
+            result["updelay"] is not None or result["downdelay"] is not None
+        ) and not result["miimon"]:
+            raise ValidationError(
+                name,
+                "the bond option downdelay or updelay is only valid with miimon enabled",
+            )
+
+        if result["num_grat_arp"] != result["num_unsol_na"]:
```

Comment:
for kernel, there are no two such options. And for NM,it will also just honor one.

No need to have both in the API.